merkle/src/store.rs

```diff
@@ -1,33 +1,74 @@
-use failure::Error;
-use merkle::Element;
+use failure::{Error, err_msg};
+use merkle::{Element, next_pow2};
 use positioned_io::{ReadAt, WriteAt};
-use std::fs::File;
+use serde::{Serialize, Deserialize};
+use std::fs::{File, OpenOptions};
+use std::path::{Path, PathBuf};
 use std::marker::PhantomData;
 use std::ops::{self, Index};
 use tempfile::tempfile;
 
 pub type Result<T> = std::result::Result<T, Error>;
 
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct StoreConfig {
+    /// A directory in which data (a merkle tree) can be persisted.
+    pub path: PathBuf,
+
+    /// A unique identifier used to help specify the on-disk store
+    /// location for this particular data.
+    pub id: String,
+
+    /// The number of merkle tree levels above the base to cache on disk.
+    pub levels: usize,
+}
+
+impl StoreConfig {
+    pub fn new(path: String, id: String, levels: usize) -> StoreConfig {
+        StoreConfig {
+            path: PathBuf::from(path),
+            id,
+            levels
+        }
+    }
+
+    // Deterministically create the data_path on-disk location from a
+    // path and specified id.
+    pub fn data_path(path: &PathBuf, id: &str) -> PathBuf {
+        const DATA_STORE_VERSION: u32 = 1;
```

Comment:
constants are best kept at the top of a file, after the imports

Reply:
No problem, was hoping it was scoped specifically the StoreConfig, but didn't think about it much further than that.  If it'll be accessible anywhere in the file scope, I'll rename it to something more descriptive.